components/ModEditFacilitySection.vue

```diff
@@ -253,16 +286,30 @@ import { validateAddressLineEn,
     validatePostalCode,
     validateWebsite,
     validateCityJa } from '~/utils/formValidations'
+import type { HealthcareProfessional } from '~/typedefs/gqlTypes'
 
 // Initialize the variable that will be used to mount the toast library
 let toast: ToastInterface
 
 const { t } = useI18n()
 
-const screenStore = useModerationScreenStore()
+const moderationScreenStore = useModerationScreenStore()
 const facilityStore = useFacilitiesStore()
+const healthcareProfessionalsStore = useHealthcareProfessionalsStore()
 
 const isFacilitySectionInitialized: Ref<boolean> = ref(false)
+const healthcareProfessionalsRelatedToFacility: Ref<string[]>
+= ref([])
+const healthcareProfessionalRelatedToFacilityFiltered: Ref<HealthcareProfessional[]> = ref([])
+
+const setHealthcareProfessionalsRelatedToFacility = () => {
+    healthcareProfessionalRelatedToFacilityFiltered.value = healthcareProfessionalsRelatedToFacility.value.flatMap(
```

Comment:
first time I've seen flatMap actually used! ....I don't think we need it though. This can just be `map()` I think?

Reply:
Correct. I used to make sure it was the correct type but if I use `as HealthcareProfessional` it works as a map

Reply:
If I do not use flatMap it doesn't remove any undefined values. This is to safely remove anything that is undefined. Otherwise the website could break on reload